### DISCUSSION TITLE
Theia browser support

### DISCUSSION THREAD

**Discussion by sakthivel-axim** [eclipse-theia/theia#11404] [Q&A]
I would like to work with the IDE through browser. 

- Please share the details to start IDE in Browser

- Does theia supports following functionalities while running in browser,

      1. Code Editor 
      2. Compiler 
      3. Build tools 
      4. Debugger 
      5. Interpreter
      6. Syntax highlighting
      7. Class Browser
      8. Object Browser 
      9. source navigation
      10. Git support
      11. Terminal
      12. Code Completion

**Comment by msujew** [comment]
Hi @sakthivel-axim,

> Does theia supports following functionalities while running in browser,

The supported feature set is the same across the browser and electron runtime. It aims to emulate the basic behavior of vscode, so it doesn't ship any compilers, build tools, interpreter, class browser or object browser. Editor support, debuggers, interpreters, syntax highlighting, source navigation and code completion support are contributed using [vscode extensions](https://theia-ide.org/docs/extensions/#vs-code-extensions) and are not part of the Theia framework itself.

> Please share the details to start IDE in Browser

Please refer to our [documentation page](https://theia-ide.org/docs/composing_applications/) to see how to build and run a browser based Theia application.

**Reply by sakthivel-axim** [reply]
Thanks @msujew . I would like to build browser IDE and run C applications. Could you please share the respective documentation link ?

Something like https://github.com/eclipse-theia/theia/discussions/8793

**Reply by msujew** [reply]
> Could you please share the respective documentation link?

@sakthivel-axim Unfortunately, we don't provide documentation for specific use cases of Theia. 

> I would like to build browser IDE and run C applications.

In short: [This](https://theia-ide.org/docs/composing_applications/) shows you how to build a browser IDE. For C language support, you'll need something like [clangd extension](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd) and to compile these files you'll have to equip your docker images with the appropriate compilers/SDKs.
